obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1071 @@
+            self.lines = LinesIterator(f.readlines())
+        return self._deserialize()
+
+    def _add_line_nb(self, message):
+        """
+        Add line number at the end of a str message.
+
+        :type message: str
+        :param message: Message for warnings or exceptions.
+        :rtype: str
+        :return: Message with line number.
+        """
+        return "%s, line %s" % (message, self.lines.line_nb)
+
+    def _warn(self, message):
+        warnings.warn(self._add_line_nb(message))
+
+    def _skip_empty_lines(self):
+        line = next(self.lines)
+        while not line or line.isspace():
+            line = next(self.lines)
+        return line
+
+    def _get_res_id(self, ident, parent=None, parent_res_id=None):
+        """
+        Create a :class:`~obspy.core.event.base.ResourceIdentifier` object.
+
+        :type ident: str
+        :param ident: Id of
+            the :class:`~obspy.core.event.base.ResourceIdentifier`.
+        :type parent: :class:`~obspy.core.event.origin.Origin`,
+            :class:`~obspy.core.event.Event` or any other object with a
+            resource_id attribute.
+        :param parent: The resource_id attribute of the parent will be
+            used as a prefix for the new
+            :class:`~obspy.core.event.base.ResourceIdentifier`.
+        :type parent_res_id: :class:`~obspy.core.event.ResourceIdentifier`
+            of the parent.
+        :param parent_res_id: :class:`~obspy.core.event.ResourceIdentifier`
+        :rtype: :class:`~obspy.core.event.ResourceIdentifier`
+        :return: ResourceIdentifier object.
+        """
+        prefix = self.res_id_prefix
+        # Put the parent id as prefix
+        # Example: smi:local/origin/351412/arrival/6389611
+        #          |        prefix        |     ident    |
+        if parent:
+            prefix = parent.resource_id.id
+        elif parent_res_id:
+            prefix = parent_res_id.id
+
+        public_id = "%s/%s" % (prefix, ident)
+        return ResourceIdentifier(public_id)
+
+    def _comment(self, text):
+        comment = Comment()
+        comment.text = text
+        comment.resource_id = ResourceIdentifier(prefix=self.res_id_prefix)
+        return comment
+
+    def _get_creation_info(self):
+        creation_info = CreationInfo(creation_time=UTCDateTime())
+        if self.agency:
+            creation_info.agency_id = self.agency
+        if self.author:
+            creation_info.author = self.author
+        return creation_info
+
+    def _check_header(self, first_line):
+        """
+        Just check some stuff in header.
+
+        :type first_line: str
+        :param first_line: First line of header.
+        """
+        line_1_pattern = 'BEGIN\sGSE2.0'
+        line_2_pattern = 'MSG_TYPE\s(REQUEST|DATA|SUBSCRIPTION|PROBLEM)'
+        line_3_pattern = 'MSG_ID\s\w{1,20}\s?\w{1,8}'
+
+        if not re.match(line_1_pattern, first_line):
+            raise SyntaxError('Wrong GSE2.0 header')
```

Comment:
I replace `SyntaxError` by a custom `GSE2BulletinSyntaxError`. I think it's cleaner than using the ValueError.